code/campaign/src/main/java/com/adobe/marketing/mobile/campaign/CampaignExtension.java

```diff
@@ -304,6 +310,16 @@ void processConfigurationResponse(final Event event) {
 
         setCampaignState(event);
 
+        // attempt to load cached rules on the first configuration event received
```

Comment:
Might be a noobie question

 Any reason we cant have the cached rules to be loaded during app launch?

Reply:
nope, good question. we can load them on app launch. the current behavior just matches the 1.x behavior but i can update it.

Reply:
I am thinking we can keep the same way how swift has implemented it

Reply:
ok, i'll load the cached rules on first config event